docs/developer-docs/security/security-best-practices/data-integrity-and-authenticity.mdx

```diff
@@ -8,19 +8,324 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 
 <MarkdownChipRow labels={["Intermediate", "Security", "Concept"]} />
 
-## Certify query responses if they are relevant for security
+## Using Certified Variables for secure queries 
 
-### Security concern
+ICP offers three modes of operations for canisters namely `update`, `query` and `composite_query`. For the sake of simplicity, we will club `composite_query` under queries for the rest of the section. A detailed overview between `update` and `query` calls can be found [here](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/call/overview/#introduction). In general, queries have fast response time but lack integrity guarantees, which makes them unfit for security critical applications. To obtain query like response times and update like certified responses, developers could make use of `certified_data` in the state tree to certify the response served in a query. Alternatively, these calls could be issued as update calls by the caller (e.g. in agent-js), but that impacts performance by taking a few seconds. Some examples of Certified Variables are asset certification in [Internet Identity](https://github.com/dfinity/internet-identity/blob/b29a6f68bbe5a49d048e12bc7a3263a9f43d080b/src/internet_identity/src/main.rs#L775-L808), [NNS dapp](https://github.com/dfinity/nns-dapp/blob/372c3562127d70c2fde059bc9c268e8ae858583e/rs/src/assets.rs#L121-L145), or the [canister signature implementation in Internet Identity](https://github.com/dfinity/internet-identity/blob/main/src/internet_identity/src/signature_map.rs).
 
-The responses to [query calls](/docs/current/references/ic-interface-spec#http-interface) are not threshold-signed by the canister and subnet. Thus, a single malicious replica or boundary node may change the data, violating its authenticity. This is especially risky if update calls, which do returned signed responses, depend on the response to query calls.
+:::tip
+ICP also provides Replica signed queries, where query responses are signed by the answering replica node but however it doesn't have the same security guarantees as `update` calls. Replica signed queries are enabled by default on both `agent-rs` and `agent-js`. You can read more about the feature [here](https://internetcomputer.org/blog/features/replica-signed-queries)
+:::
 
-### Recommendation
+### What's Certified data?
+* A canister can store a small amount of data (32 bytes) during an update call in the state tree under `certified_data` which will be certified by the subnet.
```

Comment:
Why is this info formatted as bullet points? Could this be reformatted as a normal paragraph?